src/main/app/controller/PostController.test.ts

```diff
@@ -276,6 +279,34 @@ describe('PostController', () => {
 
     expect(res.redirect).toHaveBeenCalledWith('/next-step-url');
   });
+
+  test('getEventName returns UPDATE_AOS when shouldUpdateAos returns true', async () => {
+    shouldUpdateAosMock.mockReturnValue(true);
+    const body = { gender: Gender.MALE };
+    const controller = new PostController(mockFormContent.fields);
+
+    const req = mockRequest({ body });
+    req.session.isApplicant2 = true;
+    req.session.userCase.applicationType = ApplicationType.SOLE_APPLICATION;
+    const res = mockResponse();
+    await controller.post(req, res);
+
+    expect(req.locals.api.triggerEvent).toHaveBeenCalledWith('1234', { gender: 'male' }, UPDATE_AOS);
+  });
+
+  test('getEventName should not return UPDATE_AOS when shouldUpdateAos returns false', async () => {
+    shouldUpdateAosMock.mockReturnValue(false);
+    const body = { gender: Gender.MALE };
+    const controller = new PostController(mockFormContent.fields);
+
+    const req = mockRequest({ body });
+    req.session.isApplicant2 = true;
+    req.session.userCase.applicationType = ApplicationType.SOLE_APPLICATION;
+    const res = mockResponse();
+    await controller.post(req, res);
+
+    expect(req.locals.api.triggerEvent).not.toHaveBeenCalledWith('1234', { gender: 'male' }, UPDATE_AOS);
```

Comment:
is there a test for Draft_AOS?

Reply:
https://github.com/hmcts/nfdiv-frontend/pull/2408/files#diff-c1d5768373b57e8a93385dc918e6318a47509092fed497fb43cb6bf71df2b3b4R22